src/base/intl_support.c

```diff
@@ -2726,6 +2726,36 @@ intl_strcasecmp_utf8_one_cp (const ALPHABET_DATA * alphabet, unsigned char *str1
   return 0;
 }
 
+/*
+ * intl_identifier_casecmp_for_dblinke() - compares two identifiers strings
+ *			       case insensitive excluding double quote for dblink
+ *
+ *   return: 0 if strings are equal, -1 if str1 < str2 , 1 if str1 > str2
+ *   str1(in):
+ *   str2(in):
+ *
+ * NOTE: this routine is the same as intl_identifier_casecmp
+ *       the first argument str1 may start with double quote but the second one never
+ */
+int
+intl_identifier_casecmp_for_dblink (const char *str1, const char *str2)
+{
+  int str1_size;
+  int str2_size;
+  int offset;
+
+  assert (str1 != NULL);
+  assert (str2 != NULL);
+
+  str1_size = (*str1 == '\"') ? strlen (str1) - 2 : strlen (str1);
+  str2_size = strlen (str2);
+
+  offset = (*str1 == '\"') ? 1 : 0;
+
+  return intl_identifier_casecmp_w_size (lang_id (), (unsigned char *) str1 + offset, (unsigned char *) str2, str1_size,
```

Comment:
Is it correct to compare with casecmp?


```
drop table tbl;
create table tbl( "Id" int, "id" int, id int);
insert into tbl ("Id", "id", id) values (1,2,3);
select * from tbl;
```

If you execute the example query above in Oracle, the select result item will look like the following.

```
Id	id	ID
1	2	3
```

If it was enclosed in quotation marks, it seems that the comparison should be case sensitive.

Reply:
I know, but it should be a constraint as a specification of 11.3.